src/otp-input/otp-input.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useEffect, useRef, useState } from "react";
+import { OtpInputProps } from "./types";
+import {
+    CTAButton,
+    ErrorContainer,
+    InputContainer,
+    InputField,
+    Wrapper,
+} from "./otp-input.styles";
+
+export const OtpInput = ({
+    cooldownDuration,
+    ctaProps,
+    errorMessage,
+    numOfInput,
+    onChange,
+    ...otherProps
+}: OtpInputProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const {
+        disabled,
+        onClick,
+        styleType = "secondary",
+        ...otherCtaProps
+    } = ctaProps ?? {};
+    const inputRefs = useRef<(HTMLInputElement | null)[]>([]);
+
+    const [otpValues, setOTPValues] = useState<string[]>(
+        new Array(numOfInput).fill("")
+    );
+    const [countDown, setCountDown] = useState<number>(cooldownDuration);
+    const [enableCountDownTimer, setEnableCountDownTimer] =
+        useState<boolean>(true);
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+    useEffect(() => {
+        inputRefs?.current[0]?.focus();
+        document.addEventListener("paste", handleOnPaste);
+        return () => document.removeEventListener("paste", handleOnPaste);
+    }, []);
+
+    useEffect(() => {
+        if (
+            !enableCountDownTimer ||
+            disabled ||
+            otherProps.disabled ||
+            otherProps.readOnly
+        ) {
+            return;
+        }
+        let timer = cooldownDuration;
+        setCountDown(timer);
+        const interval = setInterval(() => {
+            timer--;
+            setCountDown(timer);
+            if (timer === 0) {
+                clearInterval(interval);
+                setEnableCountDownTimer(false);
+            }
+        }, 1000);
+        return () => clearInterval(interval);
+    }, [enableCountDownTimer]);
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+    const handleOnKeyDown = (
```

Comment:
our convention is `handle<Action>` hence the word `on` is not necessary. `on` is used for the prop type to indicate a callback event